services/py-api/src/service/hackathon_service.py

```diff
@@ -53,6 +53,20 @@ async def create_participant_and_team_in_transaction(
             self._create_participant_and_team_in_transaction_callback, input_data
         )
 
+    async def create_random_participant(
+        self, input_data: ParticipantRequestBody
+    ) -> Result[Tuple[Participant, None], DuplicateEmailError | Exception]:
+        """Creates a random participant. The participant is created with team_id=None as we create the Random teams
```

Comment:
This comment is incorrect. It's created with `team_id=None` only if the user passes it as such. Either fix the type or explicitly set the `team_id=None` if you decide not to ditch the method as a whole .

Reply:
So if we leave the method we should do something like this:
```python
    async def create_random_participant(
        self, input_data: ParticipantRequestBody
    ) -> Result[Tuple[Participant, None], DuplicateEmailError | Exception]:
        """Creates a random participant. The participant is created with team_id=None as we create the Random teams
        after the registration is closed, until then random participants are not added to a team."""

        # We create a copy of the input data in order to avoid modifying it directly during runtime, as this
        # could create strange behaviour down the line.
        random_participant_data = input_data.model_copy(update={"team_name": None})
        result = await self._participant_repo.create(random_participant_data)

        if is_err(result):
            return result

        # As when first created, the random participant is not assigned to a team we return the team as None
        return Ok((result.ok_value, None))
 ```
 In order to be 100% sure that `team_name` is None, no matter what is passed to it?

Reply:
Guys remember that to access this function it passes through a check that checks if the team_id=None. Thus we dont need to assign it.

Reply:
```Python
if input_data.is_admin and input_data.team_name:
            result = await self._service.register_admin_participant(input_data)

        # TODO: Implement this when the invite participant case is done
        # elif input_data.is_admin is False and input_data.team_name:
        #     result = await self._service.register_invite_participant(input_data)

        else:
            result = await self._service.register_random_participant(input_data)
```

Reply:
Moreover
![image](https://github.com/user-attachments/assets/4af855f0-6b4b-43ac-9ad5-ec1dbd3a7df9)

Reply:
@KDimitrov333 You can roll back to the previous implementation on this one. 

You only need:
```Python
result = await self._participant_repo.create(input_data)
```

Reply:
So should I revert it to how it was before commit 1abd3aa (my last commit)?

Reply:
Yes please!